s3tests_boto3/functional/test_s3.py

```diff
@@ -9106,6 +9106,104 @@ def test_lifecycle_expiration_header_head():
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
     eq(check_lifecycle_expiration_header(response, now, 'rule1', 1), True)
 
+@attr(resource='bucket')
+@attr(method='head')
+@attr(operation='test lifecycle expiration header check earliest rule')
+@attr('lifecycle')
+@attr('lifecycle_expiration')
+def test_lifecycle_expiration_header_earliest():
+    """
+    Check for valid x-amz-expiration header on HEAD request should return the earliest date for the rule
+    """
+    bucket_name = get_new_bucket()
+    client = get_client()
+    lifecycle={
+        "Rules": [
+        {
+            "Filter": {
+                "Tag": {"Key": "key1", "Value": "tag1"}
+            },
+            "Status": "Enabled",
+            "Expiration": {
```

Comment:
should we just use the Days element to avoid this breaking somehow in future (or at least looking as if it might)?

Reply:
It is easier to check if the year 2029 than just adding days. 
It will maybe break in 2029.
Although that it is not using sleep so the object will not be deleted till the head request. 
Can we leave as it is now? 

Thanks

Reply:
@mattbenjamin I have changed it to on days based.
Is it ok?

Thanks